app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/address/enter_uk_address.scala.html

```diff
@@ -74,11 +79,17 @@
     )
   case c: AddressJourneyType.Returns.EnteringCompanyDetails =>
     if(c.draftReturn.isLeft)                        messages("address.uk.companyDetails.multipleIndirect.title")
-    else if(c.representativeType.exists(_.isLeft))  messages("address.uk.companyDetails.personalRep.title")
-    else if(c.representativeType.exists(_.isRight)) messages("address.uk.companyDetails.capacitor.title")
-    else if(c.isATrust)                             messages("address.uk.companyDetails.trust.title")
-    else if(request.userType.exists(_ === Agent))   messages("address.uk.companyDetails.agent.title")
-    else                                            messages("address.uk.companyDetails.title")
+    else
+        c.representativeType match {
+            case Some(PersonalRepresentative) => messages("address.uk.companyDetails.personalRep.title")
+            case Some(PersonalRepresentativeInPeriodOfAdmin) => messages("address.uk.companyDetails.personalRep.title")
+            case Some(Capacitor) => messages("address.uk.companyDetails.capacitor.title")
+            case None =>
+                if(c.isATrust)                                  messages("address.uk.companyDetails.trust.title")
+                else if(request.userType.exists(_ === Agent))   messages("address.uk.companyDetails.agent.title")
```

Comment:
Is it possible to use existing "@isAgent" reference here?

Reply:
done